### PR TITLE
Cleannup Max and MinLengthAttribute tests

### PR DESCRIPTION
**Pull request by hughbe**
And add some tests for multidimensional arrays, and an object that only
implements ICollection<T>

### CODE REVIEW

**Comment by hughbe** [comment]
Test Innerloop CentOS7.1 Debug Build and Test

File: src/System.ComponentModel.Annotations/tests/MinLengthAttributeTests.cs [hunk 1]
```diff
@@ -2,82 +2,86 @@
...
-        [Fact]
-        public static void GetValidationResult_validates_array_length()
+        [Theory]
+        [MemberData(nameof(Valid_TestData))]
+        public static void GetValidationResult_ValidValue_ReturnsSuccess(int length, object value)
         {
-            Assert.Equal(ValidationResult.Success, new MinLengthAttribute(0).GetValidationResult(new int[0], s_testValidationContext));
-            Assert.Equal(ValidationResult.Success, new MinLengthAttribute(12).GetValidationResult(new int[13], s_testValidationContext));
-            Assert.Equal(ValidationResult.Success, new MinLengthAttribute(16).GetValidationResult(new string[16], s_testValidationContext));
-            Assert.NotNull((new MinLengthAttribute(15).GetValidationResult(new byte[14], s_testValidationContext)).ErrorMessage);
```
Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/MinLengthAttributeTests.cs]: Where did the ErrorMessage checks go?

**Comment by stephentoub** [comment]
One comment/question, otherwise LGTM. Thanks.

File: src/System.ComponentModel.Annotations/tests/MinLengthAttributeTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/MinLengthAttributeTests.cs]: Added them back

**Comment by stephentoub** [comment]
LGTM. Thanks.

**Pull request merged**